src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9511,6 +9511,7 @@ void Unit::ScheduleAIChange()
         RestoreDisabledAI();
         PushAI(nullptr); //This could actually be PopAI() to get the previous AI but it's required atm to trigger UpdateCharmAI()
     }
+    RefreshAI();
```

Comment:
the whole point of ScheduleAIChange() is to not change it during this world update tick but on next one to avoid crashes

Reply:
Yes, refresh is unnecessary, tested without it.